raiden/api/python.py

```diff
@@ -865,19 +869,47 @@ def get_token_network_events_raiden(
         )
 
         for block_number, event in raiden_events:
-            new_event = {
-                'block_number': block_number,
-                'event': type(event).__name__,
-            }
-            if hasattr(event, 'transfer'):
-                if event.transfer.token == token_address:
-                    event.transfer = repr(event.transfer)
-            elif hasattr(event, 'token'):
-                if event.token == token_address:
-                    event.balance_proof = repr(event.balance_proof)
-            new_event.update(event.__dict__)
-            returned_events.append(new_event)
+            self._check_and_append_event(returned_events, block_number, event, token_address)
+
         returned_events.sort(key=lambda evt: evt.get('block_number'), reverse=True)
         return returned_events
 
+    def _check_and_append_event(
```

Comment:
why are you encoding the events at this layer? encoding/decoding should be done at the boundaries of the application, it should not be scattered through out the code, please just return the events here and encode them in the rest api, like the rest of the code does it.

Reply:
Done